indico/modules/events/util.py

```diff
@@ -213,6 +212,7 @@ def update_object_principals(obj, new_principals, read_access=False, full_access
         grant = {'add_roles': {role}}
         revoke = {'del_roles': {role}}
 
+    new_principals = set(new_principals)
```

Comment:
isn't it a bit more flexible and convenient to pass any iterable and then convert it to a set?